net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java

```diff
@@ -155,6 +220,176 @@ public static RuleMetadata getRuleMetadata(String name) {
     return metadata;
   }
 
+  /**
+   * Create metadata for modules not present in the previously eclipse provided metadata.
+   * Work in progress.
+   *
+   * @param moduleDetails module details fetched from checkstyle metadata
+   * @return ruleMetadata for the module
+   */
+  public static RuleMetadata createRuleMetadata(ModuleDetails moduleDetails) {
+    final String[] packageTokens = moduleDetails.getFullQualifiedName().split("\\.");
+    RuleGroupMetadata group;
+    String modulePackageName = moduleDetails.getFullQualifiedName();
+    // standard checkstyle module
+    if (modulePackageName.startsWith("com.puppycrawl.tools.checkstyle")) {
+      group = getRuleGroupMetadata(sPackageToGroupNameMap
+              .get(packageTokens[packageTokens.length - 2]));
+    }
+    // third party extension modules
+    else {
+      String lookupKey = findLookupKey(modulePackageName);
+      String ruleGroupNameString = sThirdPartyRuleGroupMap.get(lookupKey).get("name");
```

Comment:
Please no types in names, remove `String`

Reply:
Done